agent/containermetadata/manager.go

```diff
@@ -0,0 +1,159 @@
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package containermetadata
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/aws/amazon-ecs-agent/agent/config"
+	"github.com/aws/amazon-ecs-agent/agent/utils/ioutilwrapper"
+	"github.com/aws/amazon-ecs-agent/agent/utils/oswrapper"
+
+	docker "github.com/fsouza/go-dockerclient"
+)
+
+const (
+	metadataEnvironmentVariable = "ECS_CONTAINER_METADATA_FILE"
```

Comment:
Please add documentation for this variable stating that this will be set in the container's host config and this is what customers would use to locate/access metadata.

Reply:
added documentation for `ECS_CONTAINER_METADATA_FILE` in the readme, also we should discuss if we need to include more detailed documentation wrt format of the file and the specific fields.

Reply:
not including this change in favor of separate docs pr

Reply:
Why not? You can still document what this variable does `// metadataEnvironmentVariable is ...`

Reply:
i didn't realise you meant in the code earlier. fixed.

Reply:
didnt realise you meant document in the code earlier. fixed now.